examples/kubernetes/postgres-pv.yaml

```diff
@@ -0,0 +1,31 @@
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: postgres-pv-volume
+  labels:
+    type: local
+    app: postgres
+spec:
+  storageClassName: manual
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:
+    path: "/mnt/data"
```

Comment:
This looks very specific to a particular environment. I'm sure my Kubernets cluster cannot create volumes in `/mnt/data`. Why not rely dnamic provisioning which is more or less everywhere today?

Reply:
I just followed web instructions here on how to get the Postgres up and running (can't remember the source, but the PersistentVolume thing is repeating across different sources).

So, how exactly would you change this?

Reply:
I would not include the PV at all and have just the PVC without any storage class ... and let the users provide a default storage class. Most cluster or things like Minikube have some default storage class.

Reply:
I tried the following ...

Removed PersistentVolume declaration from postgres-pv.yml and removed `storageClassName` attribute from PersistentVolumeClaim.

The PVC is then automatically fulfilled by cluster:
```
NAME                STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
postgres-pv-claim   Bound    pv0051   100Gi      RWO,ROX,RWX                   5s
```

However, when Postgres starts it encounters file permissions error again:
```
$ kubectl logs postgres -f
mkdir: cannot create directory ‘/var/lib/postgresql’: Permission denied
```
Which it doesn't have a problem with when explicitly defined PV is used as is now.

Apparently filesystem permissions are set differently in the two cases.

Reply:
What environment did you used? Maybe you just need to add securit context to the Postgres deployment.

Reply:
I played some more and I seem to have found the way to get it to work.